### ISSUE TITLE
Tailing logs failed: Error response from server: Internal Server Error

### ISSUE THREAD

**Issue by gexgik** [grafana/loki#4404]
**Describe the bug**

**To Reproduce**
1. Started Loki (SHA or version)
2. Started Promtail (SHA or version) to tail '...'
3. Query: `{} term`

**Expected behavior**

**Environment:**
 - Infrastructure: laptop
 - Deployment tool: docker env

**Screenshots, Promtail config, or terminal output**
Tailing the logs doesnt work on API or logcli either.

Tried using logcli to capture the tail log.

We get the following issue:
$ logcli query --addr="http://localhost:3100" '
{filename="/var/log/docker/c7de853e3d5d5ee90f84331371f159e0fed7410fc7bd95e28f6bd353698c895a/json.log"}
' --tail
ws://localhost:3100/loki/api/v1/tail?limit=30&query=%7Bfilename%3D%22%2Fvar%2Flog%2Fdocker%2Fc7de853e3d5d5ee90f84331371f159e0fed7410fc7bd95e28f6bd353698c895a%2Fjson.log%22%7D&start=1633340565020886000
Tailing logs failed: Error response from server: Internal Server Error
(websocket: bad handshake)

If we try without --tail we get some lines of the log.
Any ideas what the issue could be?
Thanks

**Comment by m4rcu5** [comment]
Hi,

I presume you are running the `:latest` tag? You might want to change to the release version `2.3.0` which should work properly.

On the latest build, `sha256:a29339b2f05ca10f4e74b53412e5225f9a69456819869adce720530b81d26d98`, I can confirm this issue.

**Client**
```
grafana ~ # ./logcli-linux-amd64 --addr=http://127.0.0.1:3100 query --tail '{job="varlogs"}'
ws://127.0.0.1:3100/loki/api/v1/tail?limit=30&query=%7Bjob%3D%22varlogs%22%7D&start=1633419258164258946
Tailing logs failed: Error response from server: Internal Server Error
 (websocket: bad handshake)
 ```
 
**Server**
 ```
 level=error ts=2021-10-05T08:34:18.16576009Z caller=http.go:218 org_id=fake msg="Error in upgrading websocket" err="websocket: response does not implement http.Hijacker"
level=warn ts=2021-10-05T08:34:18.166160247Z caller=logging.go:72 traceID=2fb4bc2996b5b776 orgID=fake msg="GET /loki/api/v1/tail?limit=30&query=%7Bjob%3D%22varlogs%22%7D&start=1633419258164258946 (500) 956.588µs Response: \"Internal Server Error\\n\" ws: true; Connection: Upgrade; Sec-Websocket-Key: MuuvYBE1KeZOAUrloiA5RQ==; Sec-Websocket-Version: 13; Upgrade: websocket; User-Agent: Go-http-client/1.1; "
```

**Comment by gexgik** [comment]
Thank you @m4rcu5. The problem is solved for me by using 2.3.0.
